csharp-basic-topics/ValidateNumberIsPositiveOrNegative/ValidateNumberIsPositiveOrNegative/NumberValidation.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ValidateNumberIsPositiveOrNegative
+{
+    public static class NumberValidation
+    {
+        public static int IsPositiveOrNegativeUsingConditionalMethod(int number)
+        {
+            var result = 0;
+
+            if (number > 0)
+            {
+                result = 1;
+            }
+            else if (number < 0)
+            {
+                result = - 1;
+            }
+            else
+            {
+                result = 0;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingLeftShiftMethod(int number)
+        {
+            var result = 0;
+            // Using left shift operator to check the sign bit
+            if ((number & (1 << 31)) == 0)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = -1;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingRightShiftMethod(int number)
+        {
+            var result = 0;
+            // Using right shift operator to check the sign bit
+            if ((number >> 31) == 0)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = -1;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingMathAbsMethod(int number)
+        {
+            var result = 0;
+            int absoluteValue = Math.Abs(number);
+
+            if (absoluteValue == number)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = -1;
+            }
+            return result;
+        }
+        public static int IsPositiveOrNegativeUsingMathSignMethod(int number)
+        {
```

Comment:
Here directly we can return the result of Math.Sign() function.

Reply:
Refactored and updated code

Reply:
As mentioned earlier, it can be one liner as below, as Math.Sign() function takes care of 0, Positive and Negative cases:

return Math.Sign(number);

Reply:
Updated code to return Math.Sign(number);